api/controllers/cart.controller.js

```diff
@@ -1,90 +1,158 @@
+        //     const customer = await Customer.findById(customerId);
+        //     if (!customer) {
+        //         return res.status(404).json({ message: 'Customer not found' });
+        //     }
+        //     // customerName = customer.names.customer.name;
+        // }
+
+        // cart items
+        const cartItems = cart.map(item => {
+            const plant = plants.find(p => p._id.toString() === item.plantId.toString());
+            if (!plant) {
+                throw new Error(`Plant with ID ${item.plantId} not found`);
+            }
+            return {
+                plantId: plant._id,
+                name: plant.names,
+                price: plant.sellingPrice,
+                discountedPrice: plant.discountedSellingPrice || plant.sellingPrice,
+                qty: item.qty,
+                tips: plant.tips.join(', '),
+                moreInfo: plant.moreInfo,
+                tags: plant.tags.map(tag => tag.name),
+            };
+        });
+
+        // Calci
+        let totalAmount = cartItems.reduce((total, item) => total + item.discountedPrice * item.qty, 0);
+
+        let cartData;
+        if (!uuid) {
+            // Create a new cart 
+            cartData = new Cart({
+                items: cartItems,
+                totalAmount,
+                uuid: crypto.randomUUID(), // Generate UUID
+            });
+        } else {
+            // If UUID exists, update the cart
+            cartData = await Cart.findOneAndUpdate(
+                { uuid },
+                { items: cartItems, totalAmount },
+                { new: false } 
+            );
+
+            if (!cartData) {
+                return res.status(404).json({ message: 'Cart not found' });
+            }
+        }
+
+        //  offer if offerId is there
+        if (offerId) {
+            const offer = await Offer.findById(offerId);
+
+            if (!offer) {
+                return res.status(404).json({ message: 'Offer not found' });
+            }
+
+            // Check if offer is active or  not
+            if (!offer.isActive) {
+                return res.status(400).json({ message: 'Offer is not active' });
+            }
+
+            // Check if the offer is applicable to set of the array client
+            const isClient = offer.clients.includes(customerId); // clientId  frome : --------
+            if (!isClient) {
+                return res.status(403).json({ message: 'Offer not applicable for this client' });
+            }
+
+            // Check  cart total value is more than 200 
+            if (totalAmount < 200) {
+                return res.status(400).json({ message: 'Cart value must be greater than 200 to apply the offer' });
+            }
+
+            // Calculate 
+            const percentageDiscount = (totalAmount * offer.discountPercentage) / 100;
+
+            // 
+            const offerDiscount = percentageDiscount > offer.maxDiscount ? offer.maxDiscount : percentageDiscount;
```

Comment:
this is not maxDiscount, offer.upto

please check offer collection before you are using any values